PublicGitArchive/pga/cmd/cache.go

```diff
@@ -97,3 +143,9 @@ func getIndex() (io.ReadCloser, error) {
 	}
 	return gzip.NewReader(f)
 }
+
+func checkClose(name string, c io.Closer, err *error) {
```

Comment:
I'm not sure about this piece of code, tbh.
This is changing the value of the error in an almost magical way.

I understand why you're doing this, but I'd rather check have slightly more verbose code than this subtle interaction.